docs/developers/components/managing.md

```diff
@@ -0,0 +1,162 @@
+# Managing Components
+
+Harper offers several approaches to managing components that differ between local development and Harper managed instances. This page will cover the recommended methods of developing, installing, deploying, and running Harper components.
+
+## Local Development
+
+Harper is designed to be simple to run locally. Generally, Harper should be installed locally on a machine using a global package manager install (i.e. `npm i -g harperdb`).
+
+> Before continuing, ensure Harper is installed and the `harperdb` CLI is available. For more information, review the [installation guide](../../deployments/install-harperdb/README.md).
+
+When developing a component locally there are a number of ways to run it on Harper.
+
+### `dev` and `run` commands
+
+The quickest way to run a component is by using the `dev` command within the component directory.
+
+The `harperdb dev .` command will automatically watch for file changes within the component directory, and restart the Harper threads when changes are detected.
+
+The `dev` command will **not** restart the main thread; if this is a requirement, switch to using `run` instead and manually start/stop the process to execute the main thread.
+
+Stop execution for either of these processes by sending a SIGINT (generally CTRL/CMD+C) signal to the process.
+
+### Deploying to a local Harper instance
+
+Alternatively, to mimic interfacing with a hosted Harper instance, use operation commands instead.
+
+1. Start up Harper with `harperdb`
+2. _Deploy_ the component to the local instance by executing:
+    ```sh
+    harperdb deploy_component \
+      project=<name> \
+      package=<path-to-project> \
+      restart=true \
+      replicated=true
+    ```
+   - Make sure to omit the `target` option so that it _deploys_ to the Harper instance running locally
+   - The `package=<path>` option creates a symlink to the component simplifying restarts
+   - The `restart=true` option automatically restarts Harper threads after the component is deployed
+   - The `replicated=true` option is important if running on multiple threads, but optional otherwise
+3. In another process, use the `harperdb restart` command to restart the instance's threads at any time.
```

Comment:
What does "In another process" mean?

Reply:
Like another terminal - essentially I was going for "don't stop harper and run this command; instead open a new terminal and run this command"

Reply:
If the intent is another terminal, maybe "In another terminal" would work?

Reply:
done